test/speech/index.js

```diff
@@ -0,0 +1,836 @@
+          }
+        },
+        emit: function(event, data) {
+          if (event === 'data') {
+            this.pipeTo.write(data);
+          } else if (event === 'response') {
+            this.responseHandler(data);
+          }
+        },
+        once: function() {},
+        pipe: function(stream) {
+          this.pipeTo = stream;
+        }
+      };
+
+      speech.requestWritableStream = function(protoOpts) {
+        assert.strictEqual(protoOpts.service, 'Speech');
+        assert.strictEqual(protoOpts.method, 'streamingRecognize');
+
+        return mockStream;
+      };
+
+      var stream = speech.createDetectStream(request);
+      assert(stream !== mockStream);
+
+      stream.on('data', function(data) {
+        assert.deepEqual(data, [
+          {
+            transcript: expected
+          }
+        ]);
+        assert.equal(upstreamWriteCount, 2, 'two upstream writes');
+        done();
+      });
+
+      setTimeout(function() {
+        stream.write(audioFile);
+      }, 100);
+      setTimeout(function() {
+        mockStream.emit('response', 'response');
+        mockStream.emit('data', {});
+        mockStream.emit('data', {
+          results: [
+            {
+              alternatives: []
+            }
+          ]
+        });
+      }, 200);
+      setTimeout(function() {
+        mockStream.emit('data', {
+          results: [
+            {
+              alternatives: [
+                {
+                  transcript: expected
+                }
+              ]
+            }
+          ]
+        });
+      }, 300);
+    });
+  });
+
+  describe('operation', function() {
+    var REQ = {
+      name: '1234'
+    };
+
+    it('should make an operation instance', function() {
+      var operation = speech.operation('1234');
+      assert.equal(operation.name, '1234');
+    });
+
+    it('should require a name', function() {
+      assert.throws(function() {
+        speech.operation();
+      }, Error, 'A name must be specified for an operation.');
+      assert.throws(function() {
+        speech.operation({});
+      }, Error, 'A name must be specified for an operation.');
+    });
+
+    it('should accept an object', function() {
+      var operation = speech.operation({
+        name: '1234'
+      });
+      assert.equal(operation.name, '1234');
+    });
+
+    it('should decode reponse', function(done) {
+      speech.request = function(protoOpts, reqOpts, callback) {
+        callback(null, {
+          name: '1234',
+          response: {
+            typeUrl: 'foo',
+            value: 'EicKJQoeaG93IG9sZCBpcyB0aGUgQnJvb2tseW4gQnJpZGdlFdmQez8='
+          }
+        });
+      };
+      var operation = speech.operation({
+        name: '1234'
+      });
+      operation.get(function(err, operation, apiResponse) {
+        assert.ifError(err);
+        assert.equal(operation.name, '1234');
+        assert(operation.response);
+        assert(operation.response.results);
```

Comment:
I think we probably want to move these tests to a separate file (<kbd>test/speech/operation.js</kbd>)